setup.sh

```
@@ -149,8 +156,39 @@ if [ $? -ne 0 ]; then
+SESSION_DIR="$CORE_DIR/.wwebjs_auth/session"
+
+if [ ! -d "$SESSION_DIR" ]; then
+    echo "Starting the bot directly to display the WhatsApp QR code..."
+    setsid node dist/index.js &
+    NODE_PID=$!
+    echo "Waiting for WhatsApp authentication... (Scan the QR code with your WhatsApp app)"
+    # Wait for the session directory to be created
+    while [ ! -d "$SESSION_DIR" ]; do
+        sleep 1
+    done
+    local AUTH_TIMEOUT=300 elapsed=0
```

Shellcheck (0.10.0)

[error] 'local' is only valid in functions.

(SC2168)